firstvoices/backend/tests/test_apis/test_pages_api.py

```diff
@@ -2,26 +2,81 @@
 
 import pytest
 
-from backend.models.constants import Role, Visibility
-from backend.models.widget import SiteWidgetListOrder
+from backend.models import SitePage
+from backend.models.constants import AppRole, Role, Visibility
+from backend.models.widget import SiteWidget, SiteWidgetList, SiteWidgetListOrder
 from backend.tests import factories
-from backend.tests.test_apis.base_api_test import (
-    BaseReadOnlyControlledSiteContentApiTest,
-)
+from backend.tests.test_apis.base_api_test import BaseControlledSiteContentApiTest
 from backend.tests.utils import (
     setup_widget_list,
     update_widget_list_order,
     update_widget_sites,
 )
 
 
-class TestSitePageEndpoint(BaseReadOnlyControlledSiteContentApiTest):
+class TestSitePageEndpoint(BaseControlledSiteContentApiTest):
     API_LIST_VIEW = "api:sitepage-list"
     API_DETAIL_VIEW = "api:sitepage-detail"
 
+    model = SitePage
+
     def create_minimal_instance(self, site, visibility):
         return factories.SitePageFactory.create(site=site, visibility=visibility)
 
+    def get_valid_data(self, site=None):
+        widget_list = factories.SiteWidgetListWithThreeWidgetsFactory.create(site=site)
+        widgets = SiteWidget.objects.filter(sitewidgetlist_set=widget_list)
+        for widget in widgets:
+            widget.site = site
+            widget.save()
+        widget_ids = list(map(lambda x: str(x.id), widget_list.widgets.all()))
+        banner_image = factories.ImageFactory.create(site=site)
+
+        return {
+            "title": "Title",
+            "visibility": "Public",
+            "subtitle": "Subtitle",
+            "slug": "page-slug",
+            "widgets": widget_ids,
+            "banner_image": str(banner_image.id),
+            "banner_video": None,
+        }
+
+    def add_related_objects(self, instance):
+        # no related objects to add
+        pass
+
+    def assert_related_objects_deleted(self, instance):
+        # no related objects to delete
```

Comment:
The list order items should be deleted when the page is deleted, right? This would be a good place to assert that